.github/workflows/build-and-test-python-wheels.yml

```
@@ -0,0 +1,114 @@
+name: Build and Test Python Wheels
+
+on:
+  push:
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+jobs:
+  build-cccl:
+    name: Build CCCL Wheels
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/build_cuda_cccl_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      upload-artifact: true
+      upload-artifact-name: wheel-cccl-${{ matrix.python-version }}
+      upload-artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-cpu16
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  test-cccl:
+    name: Test CCCL Wheels
+    needs: build-cccl
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/test_cuda_cccl_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      needs-artifact: true
+      artifact-name: wheel-cccl-${{ matrix.python-version }}
+      artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-cpu16
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  build-cooperative:
+    name: Build Cooperative Wheels
```

xref https://github.com/NVIDIA/cccl/pull/4679#discussion_r2093688121
```suggestion
  build-cooperative:
    name: Build Cooperative Wheels
    needs: build-cccl
```